redbot/cogs/admin/admin.py

```diff
@@ -509,6 +510,40 @@ async def selfroleset_remove(self, ctx: commands.Context, *roles: SelfRole):
 
         await ctx.send(message)
 
+    @selfroleset.command(name="clear")
+    async def selfroleset_clear(self, ctx: commands.Context):
+        """Clear the list of available selfroles for this server."""
+        current_selfroles = await self.config.guild(ctx.guild).selfroles()
+
+        if not current_selfroles:
+            return await ctx.send(_("There are currently no selfroles."))
+
+        await ctx.send(
+            _("Are you sure you want to clear this server's selfrole list?") + " (yes/no)"
+        )
+        try:
+            pred = MessagePredicate.yes_or_no(ctx, user=ctx.author)
+            await ctx.bot.wait_for("message", check=pred, timeout=60)
+        except asyncio.TimeoutError:
+            await ctx.send(_("You took too long to respond."))
+            return
+        if pred.result:
+            for role in current_selfroles:
+                role = ctx.guild.get_role(role)
+                if role is None:
+                    continue
+                if not self.pass_user_hierarchy_check(ctx, role):
+                    await ctx.send(
+                        _(
+                            "I cannot let you remove {role.name} from being a selfrole because that role is higher than or equal to your highest role in the Discord hierarchy."
```

Comment:
It might be worth mentioning the whole clear operation was aborted as it is currently ambiguous as to whether this aborted the operation completely or cleared everything except the mentioned role. This sentence is a little long now so feel free ignore the suggestion and reword if you can think of something better than me.
```suggestion
                            "I cannot clear selfroles because I cannot let you remove {role.name} from being a selfrole because that role is higher than or equal to your highest role in the Discord hierarchy."
```